backend/prisma/userSeed.ts

```diff
@@ -0,0 +1,141 @@
+import prisma from '../utils/prismaClient'
+import { RelationshipStatus } from '@prisma/client'
+import crypto from 'crypto'
+const db = prisma
+
+const realUsers = [
+  {
+    email: 'test1@gmail.com',
+    firstName: 'john',
+    lastName: 'smith'
+  }, {
+    email: 'test2@gmail.com',
+    firstName: 'Xavier',
+    lastName: 'Moss'
+  }, {
+    email: 'test3@gmail.com',
+    firstName: 'Kaylee',
+    lastName: 'Ryan'
+  }, {
+    email: 'test4@gmail.com',
+    firstName: 'Scott',
+    lastName: 'Lee'
+  }, {
+    email: 'test5@gmail.com',
+    firstName: 'Carly',
+    lastName: 'Hendrix'
+  }, {
+    email: 'test6@gmail.com',
+    firstName: 'Ella',
+    lastName: 'Haas'
+  }, {
+    email: 'test7@gmail.com',
+    firstName: 'Shania',
+    lastName: 'Baxter'
+  }, {
+    email: 'test8@gmail.com',
+    firstName: 'Halle',
+    lastName: 'Pratt'
+  }, {
+    email: 'test9@gmail.com',
+    firstName: 'Miya',
+    lastName: 'Reid'
+  }, {
+    email: 'test10@gmail.com',
+    firstName: 'Wesley',
+    lastName: 'Lowery'
+  }, {
+    email: 'test111@gmail.com',
+    firstName: 'Bianca',
+    lastName: 'Lawrence'
+  }
+]
+
+const testUser = {
+  email: 'test@gmail.com',
+  firstName: 'bob',
+  lastName: 'admin'
+}
+
+const main = async (): Promise<void> => {
+  // create a test user,
+  try {
+    const salt = crypto.randomBytes(16).toString('base64');
+    crypto.pbkdf2('test', salt, 310000, 32, 'sha256', async function (err, hashedPassword) {
+      if (err !== null) throw new Error('failed to create test user')
+
+      await db.user.create({
+        data: {
+          email: testUser.email,
+          firstName: testUser.firstName,
+          lastName: testUser.lastName,
+          password: hashedPassword.toString('base64'),
+          salt
+        }
+      })
+    })
+  } catch (err) {
+    console.error(err)
+  }
+
+  // Execute 1 sec later
+  setTimeout(async () => {
```

Comment:
So I discovered that the `crypto.pbkdf2` fn takes some time to create a hash value, and afterward, create a test user record. This setTimeout solution guarantees that a test user record was created before moving on to the next step.

Reply:
while this is probably fine given this is a seed script, in reality you are mixing Callback style js programming with async/await. due to the inherently async nature of js, your line 64 begins an asynchronous execution of code but does not block the rest of this script from running. the best thing you could do here is use a promisification library such as bluebird to asyncify the crypto.pbkdf2 call and convert the callback into a promise. alternatively I think you could actually do it without al library but this presents an opportunity to look into those libraries if you wish to

Reply:
Got this taken care of by promisifying the cypto function

Reply:
i love it, good shit!